src/ffi/mod.rs

```diff
@@ -401,6 +415,15 @@ pub extern "C" fn telio_get_adapter_luid(dev: &telio) -> u64 {
     }
 }
 
+fn char_ptr_to_type<T: std::str::FromStr>(private_key: *const c_char) -> Result<T, telio_result> {
+    let cstr = unsafe { CStr::from_ptr(private_key) };
+    let private_key = cstr.to_str().map_err(|_| TELIO_RES_INVALID_STRING);
+    match private_key {
+        Ok(key) => key.parse().map_err(|_| TELIO_RES_INVALID_STRING),
+        Err(e) => Err(e),
+    }
+}
+
```

Comment:
Yep. Initially had written the method just for private_key but after going over the code realized it could be reused for other places as well. Leftover refactoring.